boot/bootutil/include/bootutil/crypto/hmac_sha256.h

```diff
@@ -22,7 +22,6 @@
     #include <stddef.h>
     #include <mbedtls/cmac.h>
     #include <mbedtls/md.h>
-    #include <mbedtls/md_internal.h>
```

Comment:
is this header inclusion removal backward compatible, or nobody else use this future but mbedtls 3.0.0 users?

Reply:
looks like not a problem

Reply:
I suggested to remove this include entirely, not just surround it with an `#if MBEDTLS_VERSION_NUMBER` because
there is nothing used in the `hmac_sha256.h` file from the `md_internal.h` header.
I also tested it without upgrading mbedtls in the `ext` folder and all the CI tests passed (no regression according to the CI).